packages/commonwealth/server/routes/contracts/updateContract.ts

```diff
@@ -0,0 +1,68 @@
+import { NextFunction } from 'express';
+import { Op } from 'sequelize';
+import { factory, formatFilename } from 'common-common/src/logging';
+import { ContractType } from 'common-common/src/types';
+import { DB } from '../../models';
+import { ContractAttributes } from '../../models/contract';
+import { TypedRequestBody, TypedResponse, success } from '../../types';
+const log = factory.getLogger(formatFilename(__filename));
+
+export const Errors = {
+  NotLoggedIn: 'Not logged in',
+  NoContractFound: 'Must provide contract id',
+  CantChangeNetwork: 'Cannot change chain network',
+  NotAdmin: 'Not an admin',
+  NoChainFound: 'Chain not found',
+};
+
+type UpdateContractReq = ContractAttributes & {
+    node_url: string;
+    address: string;
+    abi: string,
+    contractType: ContractType;
+  };
+
+type UpdateContractResp = ContractAttributes;
+
+const updateContract = async (
```

Comment:
i'm not sure we need this route yet? we don't have a UI yet for this feature or tests to know if it works as expected!